format/macho/macho.go

```diff
@@ -0,0 +1,492 @@
+package macho
+
+// https://github.com/aidansteele/osx-abi-macho-file-format-reference
+
+import (
+	"fmt"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/format/registry"
+	"github.com/wader/fq/pkg/bitio"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+func init() {
+	registry.MustRegister(decode.Format{
+		Name:        format.MACHO,
+		Description: "Mach-O is the native executable format of binaries in OS X",
+		Groups:      []string{format.PROBE},
+		DecodeFn:    machoDecode,
+	})
+}
+
+const (
+	MH_MAGIC    = 0xfeedface
+	MH_CIGAM    = 0xcefaedfe
```

Comment:
You can add  `//nolint:revive` above the const if you want to shut the linter up about this. For ELF i've also kept the original const names as i think it makes sense

Reply:
Added.